taxonomy/utils.py

```diff
@@ -211,19 +211,30 @@ def get_blacklisted_course_skills(course_key, prefetch_skills=True):
 
 def get_course_jobs(course_key):
     """
-    Get all the course jobs.
+    Get data for all course jobs.
 
     Arguments:
         course_key (str): Key of the course whose course skills need to be returned.
 
     Returns:
-        list: A list of all the course jobs.
+        list: A list of dicts where each dict contain information about a particular job.
     """
     course_skills = get_whitelisted_course_skills(course_key)
     job_skills = JobSkills.objects.select_related(
         'skill',
         'job',
+        'job__jobpostings',
     ).filter(
         skill__in=[course_skill.skill for course_skill in course_skills]
     )
-    return list({job_skill.job.name for job_skill in job_skills})
+    data = []
+    for job_skill in job_skills:
+        job_posting = job_skill.job.jobpostings_set.first()
```

Comment:
Why are we using just one job-posting instead of all job postings?

Reply:
Because the relationship between `Job` and `JobPostings` is `1-1` and there will not be multiple `JobPostings` against a `Job`. Please see https://edx-internal.slack.com/archives/CLQUG16P8/p1628106311068400